src/worker.cc

```diff
@@ -640,6 +651,11 @@ void from_json(rapidjson::Document& doc, Options::Action action, Api& api) {
   // auto_data_fix is deprecated and will be turned into
   // ignore all the things costing option. maybe remove in v4?
   if (costing_str == "auto_data_fix") {
+
+    // warning for auto data fix
+    valhalla::Info info;
+    info.add_warnings()->set_description("auto_data_fix is deprecated and will be turned to ignore all the things costing option");
```

Comment:
I'd try somewhere at the top: `auto* warnings = api->mutable_info()->mutable_warnings()` and then when you want to set a warning
```
auto* warn = warnings->Add()->set_description
```

Reply:
untested

Reply:
I've refactored this to include a function. Then I just call the function whenever I need to add a warning.